// hangul-util/dist/constant.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.thousandUnits = exports.tenUnits = exports.numberUnits = exports.WEEK_DAY = exports.NORMALIZE_JUNG = exports.NORMALIZE_JONG = exports.NORMALIZE_CHO = exports.KEY_MAPS = exports.JUNG_START_CHARCODE = exports.JUNG_HANGUL = exports.JUNG_END_CHARCODE = exports.JUNG_COMPLETE_HANGUL = exports.JOSA_LIST = exports.JONG_START_CHARCODE = exports.JONG_PERIOD = exports.JONG_HANGUL = exports.JONG_END_CHARCODE = exports.JONG_COMPLETE_HANGUL = exports.HANGUL_START_CHARCODE = exports.HANGUL_END_CHARCODE = exports.CHO_PERIOD = exports.CHO_HANGUL = void 0;
var CHO_HANGUL = ['ㄱ', 'ㄲ', 'ㄴ', 'ㄷ', 'ㄸ', 'ㄹ', 'ㅁ', 'ㅂ', 'ㅃ', 'ㅅ', 'ㅆ', 'ㅇ', 'ㅈ', 'ㅉ', 'ㅊ', 'ㅋ', 'ㅌ', 'ㅍ', 'ㅎ'];
exports.CHO_HANGUL = CHO_HANGUL;
var JUNG_HANGUL = ['ㅏ', 'ㅐ', 'ㅑ', 'ㅒ', 'ㅓ', 'ㅔ', 'ㅕ', 'ㅖ', 'ㅗ', 'ㅘ', 'ㅙ', 'ㅚ', 'ㅛ', 'ㅜ', 'ㅝ', 'ㅞ', 'ㅟ', 'ㅠ', 'ㅡ', 'ㅢ', 'ㅣ'];
exports.JUNG_HANGUL = JUNG_HANGUL;
var JONG_HANGUL = ['', 'ㄱ', 'ㄲ', 'ㄳ', 'ㄴ', 'ㄵ', 'ㄶ', 'ㄷ', 'ㄹ', 'ㄺ', 'ㄻ', 'ㄼ', 'ㄽ', 'ㄾ', 'ㄿ', 'ㅀ', 'ㅁ', 'ㅂ', 'ㅄ', 'ㅅ', 'ㅆ', 'ㅇ', 'ㅈ', 'ㅊ', 'ㅋ', 'ㅌ', 'ㅍ', 'ㅎ'];
exports.JONG_HANGUL = JONG_HANGUL;
var JUNG_COMPLETE_HANGUL = {
  ㅘ: 'ㅗㅏ',
  ㅙ: 'ㅗㅐ',
  ㅚ: 'ㅗㅣ',
  ㅝ: 'ㅜㅓ',
  ㅞ: 'ㅜㅔ',
  ㅟ: 'ㅜㅣ',
  ㅢ: 'ㅡㅣ'
};
exports.JUNG_COMPLETE_HANGUL = JUNG_COMPLETE_HANGUL;
var JONG_COMPLETE_HANGUL = {
  ㄳ: 'ㄱㅅ',
  ㄵ: 'ㄴㅈ',
  ㄶ: 'ㄴㅎ',
  ㄺ: 'ㄹㄱ',
  ㄻ: 'ㄹㅁ',
  ㄼ: 'ㄹㅂ',
  ㄽ: 'ㄹㅅ',
  ㄾ: 'ㄹㅌ',
  ㄿ: 'ㄹㅍ',
  ㅀ: 'ㄹㅎ',
  ㅄ: 'ㅂㅅ'
};
exports.JONG_COMPLETE_HANGUL = JONG_COMPLETE_HANGUL;
var JUNG_START_CHARCODE = JUNG_HANGUL[0].charCodeAt(0);
exports.JUNG_START_CHARCODE = JUNG_START_CHARCODE;
var JUNG_END_CHARCODE = JUNG_HANGUL[JUNG_HANGUL.length - 1].charCodeAt(0);
exports.JUNG_END_CHARCODE = JUNG_END_CHARCODE;
var JONG_START_CHARCODE = JONG_HANGUL[1].charCodeAt(0);
exports.JONG_START_CHARCODE = JONG_START_CHARCODE;
var JONG_END_CHARCODE = JONG_HANGUL[JONG_HANGUL.length - 1].charCodeAt(0);
exports.JONG_END_CHARCODE = JONG_END_CHARCODE;
var HANGUL_START_CHARCODE = '가'.charCodeAt(0);
exports.HANGUL_START_CHARCODE = HANGUL_START_CHARCODE;
var HANGUL_END_CHARCODE = '힣'.charCodeAt(0);
exports.HANGUL_END_CHARCODE = HANGUL_END_CHARCODE;
var CHO_PERIOD = '까'.charCodeAt(0) - '가'.charCodeAt(0);
exports.CHO_PERIOD = CHO_PERIOD;
var JONG_PERIOD = '개'.charCodeAt(0) - '가'.charCodeAt(0);

// use covertKey function
exports.JONG_PERIOD = JONG_PERIOD;
var KEY_MAPS = {
  ㅂ: "q",
  ㅃ: "Q",
  ㅈ: "w",
  ㅉ: "W",
  ㄷ: "e",
  ㄸ: "E",
  ㄱ: "r",
  ㄲ: "R",
  ㅅ: "t",
  ㅆ: "T",
  ㅛ: "y",
  ㅕ: "u",
  ㅑ: "i",
  ㅐ: "o",
  ㅒ: "O",
  ㅔ: "p",
  ㅖ: "P",
  ㅁ: "a",
  ㄴ: "s",
  ㅇ: "d",
  ㄹ: "f",
  ㅎ: "g",
  ㅗ: "h",
  ㅓ: "j",
  ㅏ: "k",
  ㅣ: "l",
  ㅋ: "z",
  ㅌ: "x",
  ㅊ: "c",
  ㅍ: "v",
  ㅠ: "b",
  ㅜ: "n",
  ㅡ: "m"
};

// use formatNumber function
exports.KEY_MAPS = KEY_MAPS;
var numberUnits = ["", "일", "이", "삼", "사", "오", "육", "칠", "팔", "구"];
exports.numberUnits = numberUnits;
var tenUnits = ["", "십", "백", "천"];
exports.tenUnits = tenUnits;
var thousandUnits = ["", "만", "억", "조", "경", "해"];

// use formatDate function
exports.thousandUnits = thousandUnits;
var WEEK_DAY = ["일", "월", "화", "수", "목", "금", "토"];

// use josa function
exports.WEEK_DAY = WEEK_DAY;
var JOSA_LIST = {
  이: "이/가",
  가: "이/가",
  을: "을/를",
  를: "을/를",
  은: "은/는",
  는: "은/는",
  으로: "으로/로",
  로: "으로/로",
  와: "와/과",
  과: "와/과",
  이나: "이나/나",
  나: "이나/나",
  이에: "이에/에",
  에: "이에/에",
  이란: "이란/란",
  란: "이란/란"
};

// use normalize function
exports.JOSA_LIST = JOSA_LIST;
var NORMALIZE_CHO = ["g", "gg", "n", "d", "dd", "r", "m", "b", "bb", "s", "ss", "", "j", "jj", "c", "k", "t", "p", "h"];
exports.NORMALIZE_CHO = NORMALIZE_CHO;
var NORMALIZE_JUNG = ["a", "ae", "ya", "yae", "eo", "e", "yeo", "ye", "o", "wa", "wae", "oe", "yo", "u", "weo", "we", "wi", "yu", "eu", "eui", "i"];
exports.NORMALIZE_JUNG = NORMALIZE_JUNG;
var NORMALIZE_JONG = ["", "g", "gg", "gs", "n", "nj", "nh", "d", "r", "rk", "rm", "rb", "rs", "rt", "rp", "rh", "m", "b", "bs", "s", "ss", "ng", "j", "c", "k", "t", "p", "h"];
exports.NORMALIZE_JONG = NORMALIZE_JONG;
