// hangul-util/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  isCho: true,
  isChoByGroups: true,
  isJung: true,
  isJungByGroups: true,
  isJong: true,
  isJongByGroups: true,
  isHangul: true,
  isHangulByGroups: true,
  divideByJong: true,
  divideByJung: true,
  divideHangul: true,
  divideHangulByGroups: true,
  combineHangul: true,
  combineByJung: true,
  combineByJong: true
};
Object.defineProperty(exports, "combineByJong", {
  enumerable: true,
  get: function get() {
    return _combine.combineByJong;
  }
});
Object.defineProperty(exports, "combineByJung", {
  enumerable: true,
  get: function get() {
    return _combine.combineByJung;
  }
});
Object.defineProperty(exports, "combineHangul", {
  enumerable: true,
  get: function get() {
    return _combine.combineHangul;
  }
});
Object.defineProperty(exports, "divideByJong", {
  enumerable: true,
  get: function get() {
    return _divide.divideByJong;
  }
});
Object.defineProperty(exports, "divideByJung", {
  enumerable: true,
  get: function get() {
    return _divide.divideByJung;
  }
});
Object.defineProperty(exports, "divideHangul", {
  enumerable: true,
  get: function get() {
    return _divide.divideHangul;
  }
});
Object.defineProperty(exports, "divideHangulByGroups", {
  enumerable: true,
  get: function get() {
    return _divide.divideHangulByGroups;
  }
});
Object.defineProperty(exports, "isCho", {
  enumerable: true,
  get: function get() {
    return _isCho.isCho;
  }
});
Object.defineProperty(exports, "isChoByGroups", {
  enumerable: true,
  get: function get() {
    return _isCho.isChoByGroups;
  }
});
Object.defineProperty(exports, "isHangul", {
  enumerable: true,
  get: function get() {
    return _isHangul.isHangul;
  }
});
Object.defineProperty(exports, "isHangulByGroups", {
  enumerable: true,
  get: function get() {
    return _isHangul.isHangulByGroups;
  }
});
Object.defineProperty(exports, "isJong", {
  enumerable: true,
  get: function get() {
    return _isJong.isJong;
  }
});
Object.defineProperty(exports, "isJongByGroups", {
  enumerable: true,
  get: function get() {
    return _isJong.isJongByGroups;
  }
});
Object.defineProperty(exports, "isJung", {
  enumerable: true,
  get: function get() {
    return _isJung.isJung;
  }
});
Object.defineProperty(exports, "isJungByGroups", {
  enumerable: true,
  get: function get() {
    return _isJung.isJungByGroups;
  }
});
var _isCho = require("./isCho");
var _isJung = require("./isJung");
var _isJong = require("./isJong");
var _isHangul = require("./isHangul");
var _divide = require("./divide");
var _combine = require("./combine");
var _formatDate = require("./formatDate");
Object.keys(_formatDate).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _formatDate[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _formatDate[key];
    }
  });
});
var _formatNumber = require("./formatNumber");
Object.keys(_formatNumber).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _formatNumber[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _formatNumber[key];
    }
  });
});
var _josa = require("./josa");
Object.keys(_josa).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _josa[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _josa[key];
    }
  });
});
var _banmal = require("./banmal");
Object.keys(_banmal).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _banmal[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _banmal[key];
    }
  });
});
var _encode = require("./encode");
Object.keys(_encode).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _encode[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _encode[key];
    }
  });
});
var _distance = require("./distance");
Object.keys(_distance).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _distance[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _distance[key];
    }
  });
});
var _includesByCho = require("./includesByCho");
Object.keys(_includesByCho).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _includesByCho[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _includesByCho[key];
    }
  });
});
var _sortHangul = require("./sortHangul");
Object.keys(_sortHangul).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _sortHangul[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _sortHangul[key];
    }
  });
});
var _getLocal = require("./getLocal");
Object.keys(_getLocal).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _getLocal[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _getLocal[key];
    }
  });
});
var _convertKey = require("./convertKey");
Object.keys(_convertKey).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _convertKey[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _convertKey[key];
    }
  });
});
var _normalize = require("./normalize");
Object.keys(_normalize).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _normalize[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _normalize[key];
    }
  });
});
var _utils = require("./utils");
Object.keys(_utils).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _utils[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _utils[key];
    }
  });
});
